internal/internal_workflow_testsuite.go

```diff
@@ -2363,6 +2367,32 @@ func (env *testWorkflowEnvironmentImpl) ExecuteNexusOperation(params executeNexu
 	}
 	env.runningNexusOperations[seq] = handle
 
+	var opID string
+	if params.options.ScheduleToCloseTimeout > 0 {
+		env.NewTimer(
+			params.options.ScheduleToCloseTimeout,
+			TimerOptions{},
+			func(result *commonpb.Payloads, err error) {
+				timeoutErr := env.failureConverter.FailureToError(nexusOperationFailure(
+					params,
+					opID,
+					&failurepb.Failure{
+						Message: "operation timed out",
+						FailureInfo: &failurepb.Failure_TimeoutFailureInfo{
+							TimeoutFailureInfo: &failurepb.TimeoutFailureInfo{
+								TimeoutType: enumspb.TIMEOUT_TYPE_SCHEDULE_TO_CLOSE,
+							},
+						},
+					},
+				))
+				env.postCallback(func() {
+					handle.startedCallback(opID, timeoutErr)
```

Comment:
There's no need to send the operation ID if the operation timed out, the startedCallback should have already been called with the operation ID at this point.